Implement various translation features requested by Novachen

asarium opened the pull request:
This fixes or implements a few things requested in #1730.

New features:
- Alternative name for medals that is only used for displaying their name. This allows them to be translated without causing issues with the medal SEXPs.
- A game_settings option for disabling all currently built-in translations. Since the parts of the engine where this was used have been updated with proper translation support, the built-in translations are not needed anymore. They can be disabled with a game_settings option so that backwards compatibility is not broken.

Goober5000 commented on code/stats/medals.cpp:
```diff
@@ -609,7 +620,8 @@ void blit_label(char *label, int num)
 	// translate medal names before displaying
 	// can't translate in table cuz the names are used in comparisons
-	if (Lcl_gr) {
+	// Medals no have alternate display names so this code can be disabled in the mod table
```
This comment has a typo

Goober5000 commented on code/stats/medals.h:
```diff
@@ -46,6 +47,8 @@ class medal_stuff
 	medal_stuff(const medal_stuff &m);
 	medal_stuff& operator=(const medal_stuff &m);
 
+	const char* get_display_string();
```
Medals have `get_display_string` while weapons have `get_display_name`.  This may not matter but I thought I should mention it.

Goober5000 commented:
I notice this also fixes the weapon name in targetbox bug that someone mentioned.  (I can't find it on the issue list so maybe it was never reported as an issue.)

NovachenFS2 commented:
> Add convencience function for retrieving weapon alt name

Even i know only one situation where this name appears and this is in the Targeting Box for Bombs. But now the translated name is visible there. So it seems to be working :)

> Add alternative display name for medals 

This one actually did not work. Regardless of $Alt Name with XSTR in the medals.tbl, the name appear untranslated in the Mission Debriefing aswell in the Medal Case.

> Add option for disabling built-in translations

This looked good.

> Disable hard-coded "cluster bomb" hack 

The translated name of the Cluster Bomb appears in the weapon gauge now, so this one is working also.

NovachenFS2 commented:
The Alternate Name appears in the Medal Case, now. But in the debriefing of a mission the $Name is still displayed.

asarium commented:
Slight oversight, should be fixed now.

Goober5000 commented:
If @NovachenFS2 if happy with these latest changes, I'm happy to merge the PR.

NovachenFS2 commented:
> Add convencience function for retrieving weapon alt name 

I actually saw, that the Ballistic Weapon Gauge is displaying the $Name also, instead the $Alt Name. I think this still fits to this PR, too? Otherwise i make a new issue for that.

If there is a fix for the wrong medal name in the debriefing (do not see an updated file here?) then i did not test it yet.

asarium commented:
I fixed the `$Alt Name` usage for the ballistic weapon gauge.

The fix for the debriefing medal name should be in this PR. I added the changes to the original medals commit so that may be why you haven't seen a new commit for that.

NovachenFS2 commented:
Ah okay.

The medals are working now as i had that in mind.
The $Alt Name is displayed in the Ballistic Gauge as well, which is also good.

But this time i had an oversight by myself 😊 . In the Targetbox the $Name of the Bomb is still displayed and not the $Alt Name.

asarium commented:
Ok, I'll take a look at it tomorrow.

asarium commented:
I went through all occurrences of `name` and replaced them with the new function where ever the name was shown to the user. That should finally cover all places in the code where the alternate name should be shown.

NovachenFS2 commented:
Yes, these changes are working and the $Alt Name where shown everywhere. So everything is working now. Thank you, that is a massive help for translations. :)

asarium commented:
@Goober5000 FYI, I made some additional changes since the last code review.

Goober5000 commented:
Looks good and Novachen is happy.  Merging!
